url-builder/src/main/java/com/palominolabs/http/url/UrlBuilder.java

```diff
@@ -366,11 +355,52 @@ public String toUrlString() throws CharacterCodingException {
             buf.append("?");
             buf.append(unstructuredQueryEncoder.encode(unstructuredQuery));
         }
+    }
 
+    private void appendFragment(StringBuilder buf) throws MalformedInputException, UnmappableCharacterException {
         if (fragment != null) {
             buf.append('#');
             buf.append(fragmentEncoder.encode(fragment));
         }
+    }
+
+    public URL toUrl() throws CharacterCodingException, MalformedURLException {
```

Comment:
We should probably find a way to not throw MalformedURLException. Wrap in RuntimeException maybe? Or our own type? If we generate a malformed URL that seems like an error in this library